From 4ef6cd514a9eead377ae8adb06f222a67917b3ea Mon Sep 17 00:00:00 2001
From: Jenny TC <jenny.tc@intel.com>
Date: Thu, 18 Dec 2014 09:21:02 +0530
Subject: [PATCH 1/1] pmic_ccsm: Enable only BZONE interrupt

PMIC CCSM need to handle only BZONE interrupt in the
THRMIRQ1 register. Other interrupts (BATTCRITHOTCOLD
and BATCRIT) are not needed to implement the charging
functionality. These interrupts are level triggered -
so once triggered they remain high till the condition
recovers. This creates an overhead since they need to be
masked as soon as they are triggered and need to poll
to see if they are recovered or not. For charging the
battery zone interrupt is good enough which tell the
soc the current battery temperature zone. The charging
can be enabled/disabled based on this.

Change-Id: Ic8c893280dd7de5415bae5c5df2431f83a26f209
Signed-off-by: Jenny TC <jenny.tc@intel.com>
---
 drivers/power/intel_pmic_ccsm.c | 3 ++-
 drivers/power/intel_pmic_ccsm.h | 2 +-
 2 files changed, 3 insertions(+), 2 deletions(-)

diff --git a/drivers/power/intel_pmic_ccsm.c b/drivers/power/intel_pmic_ccsm.c
index a490da92cd46..a10ab7944f24 100644
--- a/drivers/power/intel_pmic_ccsm.c
+++ b/drivers/power/intel_pmic_ccsm.c
@@ -1683,7 +1683,8 @@ static int pmic_chrgr_probe(struct platform_device *pdev)
 		}
 	}
 
-	pmic_write_reg(chc.reg_map->pmic_mthrmirq1, MTHRMIRQ1_CCSM_MASK);
+	intel_soc_pmic_writeb(chc.reg_map->pmic_mthrmirq1,
+						~MTHRMIRQ1_CCSM_MASK);
 
 	ret = pmic_check_initial_events();
 	if (ret)
diff --git a/drivers/power/intel_pmic_ccsm.h b/drivers/power/intel_pmic_ccsm.h
index 8f0542043500..fc7808f22e26 100644
--- a/drivers/power/intel_pmic_ccsm.h
+++ b/drivers/power/intel_pmic_ccsm.h
@@ -237,7 +237,7 @@
 #define TT_CHGRENVAL_ADDR		0X3C
 #define TT_CHGRDISVAL_ADDR		0X3D
 
-#define MTHRMIRQ1_CCSM_MASK		0x90
+#define MTHRMIRQ1_CCSM_MASK		0x80
 #define MTHRMIRQ2_CCSM_MASK		0x3
 
 /*Interrupt registers*/
-- 
1.9.1

